### ISSUE TITLE
seg fault on terraform plan for `auth0_guardian` resource

### ISSUE THREAD

**Issue by acookin** [auth0/terraform-provider-auth0#296]
### Checklist

- [X] I have looked into the [README](https://github.com/auth0/terraform-provider-auth0#readme) and have not found a suitable solution or answer.
- [X] I have looked into the [documentation](https://registry.terraform.io/providers/auth0/auth0/latest/docs) and have not found a suitable solution or answer.
- [X] I have searched the [issues](https://github.com/auth0/terraform-provider-auth0/issues) and have not found a suitable solution or answer.
- [X] I have searched the [Auth0 Community](https://community.auth0.com/c/sdks/5) forums and have not found a suitable solution or answer.
- [X] I agree to the terms within the [Auth0 Code of Conduct](https://github.com/auth0/open-source-template/blob/master/CODE-OF-CONDUCT.md).

### Description

We're currently experiencing a seg fault error attempting to plan a `auth0_guardian` with version `v0.30.2`.

I also commented out our guardian resource, and ran into the same error when trying to plan a `auth0_role` resource.

Details of stack trace:

```
╷
│ Error: Plugin did not respond
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ReadResource call. The plugin logs may contain more details.
╵
╷
│ Error: Request cancelled
│ 
│   with module.auth0.auth0_email.my_email_provider[0],
│   on ../modules/auth0/main.tf line 9, in resource "auth0_email" "my_email_provider":
│    9: resource "auth0_email" "my_email_provider" {
│ 
│ The plugin.(*GRPCProvider).UpgradeResourceState request was cancelled.
╵
╷
│ Error: Plugin did not respond
│ 
│   with module.auth0.auth0_role.admin,
│   on ../modules/auth0/roles.tf line 1, in resource "auth0_role" "admin":
│    1: resource "auth0_role" "admin" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ValidateResourceConfig call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-auth0_v0.30.2 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10115d834]

goroutine 46 [running]:
github.com/auth0/terraform-provider-auth0/auth0.flattenAuth0Options(0x14000403e00)
        github.com/auth0/terraform-provider-auth0/auth0/resource_auth0_guardian.go:370 +0x54
github.com/auth0/terraform-provider-auth0/auth0.readGuardian({0x10144f590?, 0x140003be390?}, 0x101c8d3c8?, {0x10138dec0?, 0x14000403e00})
        github.com/auth0/terraform-provider-auth0/auth0/resource_auth0_guardian.go:316 +0x32c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).read(0x140003b3ea0, {0x10144f590, 0x140003be390}, 0xd?, {0x10138dec0, 0x14000403e00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/resource.go:724 +0xec
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).RefreshWithoutUpgrade(0x140003b3ea0, {0x10144f590, 0x140003be390}, 0x14000422820, {0x10138dec0, 0x14000403e00})
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/resource.go:1015 +0x46c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ReadResource(0x140002952a8, {0x10144f4e8?, 0x14000430240?}, 0x140004302c0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.14.0/helper/schema/grpc_provider.go:613 +0x40c
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ReadResource(0x1400040e000, {0x10144f590?, 0x140000f0e10?}, 0x140006420c0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:746 +0x3b8
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ReadResource_Handler({0x101415080?, 0x1400040e000}, {0x10144f590, 0x140000f0e10}, 0x14000642060, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:349 +0x174
google.golang.org/grpc.(*Server).processUnaryRPC(0x140002ac8c0, {0x101452028, 0x140001024e0}, 0x14000185200, 0x140003dce70, 0x10190c950, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1282 +0xb3c
google.golang.org/grpc.(*Server).handleStream(0x140002ac8c0, {0x101452028, 0x140001024e0}, 0x14000185200, 0x0)
        google.golang.org/grpc@v1.45.0/server.go:1619 +0x840
google.golang.org/grpc.(*Server).serveStreams.func1.2()
        google.golang.org/grpc@v1.45.0/server.go:921 +0x88
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.45.0/server.go:919 +0x298

Error: The terraform-provider-auth0_v0.30.2 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

### Expectation

Even if this resource is misconfigured, I would expect this provider not to seg fault.

However, I have not changed this resource between plans and it is now reliably failing.

### Reproduction

1. Given i install the provider at version `v0.30.2`
2. When I try to create an `auth0_guardian` or `auth0_role` resource
3. I get a seg fault

### Auth0 Terraform Provider version

auth0_guardian

### Terraform version

v1.2.7

**Comment by osdiab** [comment]
I too am experiencing this at my company.

**Comment by sergiught** [comment]
Hey folks, thanks for reporting this. Since version 0.30.2 we've pushed a lot of bug fixes and new features within the provider specially targeting the resources mentioned above. Could you please upgrade to the latest version and check if you're still experiencing this issue? If this still persists we're going to need to take a look at your terraform configuration and a verbose output of terraform apply with `TF_LOG=debug terraform apply`.

**Comment by osdiab** [comment]
I am experiencing this through Pulumi, which appears to use 0.34.0, not sure if 0.35.0 will fix it but yeah. Worth mentioning that this only started happening today, which I noticed when our CI started failing, been using it for a while without any segfaults.

**Comment by sergiught** [comment]
@osdiab Would you be able to run pulumni with TF_LOG=debug at the beginning? I think the nested provider should be able to pick it up, and it would be also great to have the stacktrace posted here as well. As the stack trace within this issue crashes at `github.com/auth0/terraform-provider-auth0/auth0/resource_auth0_guardian.go:370` which has been fixed in the latest version of the provider by using `template.GetEnrollmentMessage()` that correctly checks for nil objects.

**Comment by osdiab** [comment]
This is all that came out:

```
Diagnostics:
  auth0:index:Guardian (default):
    error: Preview failed: error reading from server: EOF
 
  pulumi:pulumi:Stack (web-app-canary):
    panic: runtime error: invalid memory address or nil pointer dereference
    [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1ce5dfb]
    goroutine 13 [running]:
    github.com/auth0/terraform-provider-auth0/auth0.flattenAuth0Options(0xc0005b9400)
    	/home/runner/go/pkg/mod/github.com/auth0/terraform-provider-auth0@v0.29.0/auth0/resource_auth0_guardian.go:367 +0x5b
    github.com/auth0/terraform-provider-auth0/auth0.readGuardian(0xc00058d6b0, {0x1eac9e0, 0xc0005b9400})
    	/home/runner/go/pkg/mod/github.com/auth0/terraform-provider-auth0@v0.29.0/auth0/resource_auth0_guardian.go:313 +0x2b1
    github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Resource).Refresh(0xc000575320, 0xc0005c6140, {0x1eac9e0, 0xc0005b9400})
    	/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/resource.go:516 +0x1df
    github.com/hashicorp/terraform-plugin-sdk/helper/schema.(*Provider).Refresh(0xc000160780, 0xc00095f690, 0x200cf4a)
    	/home/runner/go/pkg/mod/github.com/hashicorp/terraform-plugin-sdk@v1.16.1/helper/schema/provider.go:333 +0x65
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfshim/sdk-v1.v1Provider.Refresh({0xc0001ee0a0}, {0x200cf4a, 0xc000161200}, {0x22b97b8, 0xc0002d08b0})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.20.0/pkg/tfshim/sdk-v1/provider.go:102 +0x99
    github.com/pulumi/pulumi-terraform-bridge/v3/pkg/tfbridge.(*Provider).Read(0xc0004aafc0, {0x22b8788, 0xc0002ca2a0}, 0xc0005c6000)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi-terraform-bridge/v3@v3.20.0/pkg/tfbridge/provider.go:941 +0x651
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Read_Handler.func1({0x22b8788, 0xc0002ca2a0}, {0x1f85720, 0xc0005c6000})
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.27.0/proto/go/provider.pb.go:2647 +0x7b
    github.com/grpc-ecosystem/grpc-opentracing/go/otgrpc.OpenTracingServerInterceptor.func1({0x22b8788, 0xc0002ca000}, {0x1f85720, 0xc0005c6000}, 0xc00030a020, 0xc00081c0a8)
    	/home/runner/go/pkg/mod/github.com/grpc-ecosystem/grpc-opentracing@v0.0.0-20180507213350-8e809c8a8645/go/otgrpc/server.go:57 +0x425
    github.com/pulumi/pulumi/sdk/v3/proto/go._ResourceProvider_Read_Handler({0x1fc6de0, 0xc0004aafc0}, {0x22b8788, 0xc0002ca000}, 0xc00009c300, 0xc0005282c0)
    	/home/runner/go/pkg/mod/github.com/pulumi/pulumi/sdk/v3@v3.27.0/proto/go/provider.pb.go:2649 +0x138
    google.golang.org/grpc.(*Server).processUnaryRPC(0xc0005561c0, {0x22cf8c0, 0xc000162600}, 0xc0005d0480, 0xc000881110, 0x2c9eed8, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.37.0/server.go:1217 +0xc8f
    google.golang.org/grpc.(*Server).handleStream(0xc0005561c0, {0x22cf8c0, 0xc000162600}, 0xc0005d0480, 0x0)
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.37.0/server.go:1540 +0xa2a
    google.golang.org/grpc.(*Server).serveStreams.func1.2()
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.37.0/server.go:878 +0x98
    created by google.golang.org/grpc.(*Server).serveStreams.func1
    	/home/runner/go/pkg/mod/google.golang.org/grpc@v1.37.0/server.go:876 +0x294
 
    error: preview failed
```

**Comment by osdiab** [comment]
oh though i see the version showing up in the logs is even older. that's weird.

**Comment by sergiught** [comment]
Hey @osdiab indeed in your logs above v0.29.0 of the provider is being used. Just as a sanity check are you using https://github.com/pulumi/pulumi-auth0/releases/tag/v2.12.0 of pulumni-auth0?

**Comment by osdiab** [comment]
Yes, I am using it through their NPM package, v2.12.0.

**Comment by sergiught** [comment]
Hey folks, unfortunately we'll have to close this issue down as the issue you're experiencing was already fixed in the latest version of the provider (the guardian improvements and fixes came with [v0.33.0](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.33.0)). 

In your case @acookin you just need to upgrade to version [0.36.0](https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.36.0) paying attention to the breaking changes introduced in https://github.com/auth0/terraform-provider-auth0/releases/tag/v0.32.0. 

And in your case @osdiab you need to use a version of pulumni that uses the latest version of our provider as well (and not v0.29.0).
